src/util/Iterators.h

```diff
@@ -289,6 +294,23 @@ class InputRangeFromGet {
   // Get the next value and store it.
   void getNextAndStore() { storage_ = get(); }
 
+  static constexpr bool hasDetails = !std::is_same_v<Details, NoDetails>;
+  CPP_member auto details() -> CPP_ret(Details&)(requires hasDetails) {
+    return std::holds_alternative<Details>(m_details)
+               ? std::get<Details>(m_details)
+               : *std::get<Details*>(m_details);
+  }
+
+  void setDetailsPointer(Details* pointer) requires hasDetails {
+    AD_CONTRACT_CHECK(pointer != nullptr);
+    m_details = pointer;
+  }
+
+  using DetailStorage =
+      std::conditional_t<hasDetails, std::variant<Details, Details*>, Details>;
+  // If the `Details` type is empty, we don't need it to occupy any space.
+  [[no_unique_address]] DetailStorage m_details{};
```

Comment:
I think we have this exact code now twice for the Get rnage and for the TypeErased range.

Unify this (implement a templated base class or a CRTP-style mixin from which we get the details functionality)

Reply:
added crtp style [here](https://github.com/ad-freiburg/qlever/pull/2236/commits/e9d87649bd880cfff1e1e3246bd0d10a7188ff32)